app/controllers/users_controller.rb

```diff
@@ -1,7 +1,66 @@
 class UsersController < ApplicationController
 
-  def list
+  def show
     @users = User.all
   end
-  alias :index :list
+  alias :index :show
+
+  def new
+  end
+
+  def create
+    @user = User.create(user_params)
+    redirect_to '/'
+  end
+
+  def trim
+    user = User.find(params[:id])
+    user.update_attribute(:trimmed, true)
+    user.save
+    render :json => user
+  end
+
+  def massage_and_trim
+    user = User.find(params[:id])
+    user.update_attributes(:trimmed => true, :massaged => true)
+    user.save
+    render :json => user
+  end
+
+  def shave
+    user = User.find(params[:id])
+    user.update_attribute(:shaved, true)
+    user.save
+    render :json => user
+  end
+
+  def reset
+    user = User.find(params[:id])
+    if user.trimmed == true && user.massaged = true
```

Comment:
Seems like this logic is unecessary as all you care about is the result which is that the user be NOT trimmed and NOT massaged at the end.  Wouldn't just updating both of those attributes to 'false' regardless of their current state achieve the desired outcome?